### ISSUE TITLE
AppCenter codepush command failing on macOS

### ISSUE THREAD

**Issue by kartikk** [microsoft/appcenter-cli#1218]
### Problem
The appcenter CLI seems to be looking for an item that does not exist in the keychain. Login was done successfully using `appcenter login --disable-telemetry --token <codepush_access_code>`

### Command
```
appcenter codepush release-react --disabled --target-binary-version <version> --app <app> --output-dir <dir> --deployment-name <name> --disable-duplicate-release-error --description <description>
```

### Error
```
Error: Command 'appcenter codepush release-react --disabled --target-binary-version <version> --app <app> --output-dir <dir> --deployment-name <name> --disable-duplicate-release-error --description <description>' 
failed with exception "Command failed: /usr/bin/security find-generic-password -a <username> -s mobile-center-cli -g
security: SecKeychainSearchCopyNext: The specified item could not be found in the keychain.
```

### Enviornment
```
Appcenter CLI - 2.7.3
OS - macOS(running on CircleCI)
XCode - 12.1.0
```

**Comment by mike-odom** [comment]
Same issue here. I'm not sure if it's related, but I just upgrade to 2.7.4 after having not used this system for a month.

`security: SecKeychainSearchCopyNext: The specified item could not be found in the keychain.`

The command being run by `appcenter` is `/usr/bin/security find-generic-password -a [my-user-info-redacted] -s mobile-center-cli -g`
And running `/usr/bin/security find-generic-password -a [my-user-info-redacted]  -g` shows me:

```
keychain: "/Users/UserName/Library/Keychains/login.keychain-db"
version: 512
class: "genp"
attributes:
    0x00000007 <blob>="appcenter-cli"
    0x00000008 <blob>=<NULL>
    ...
```

Is that `mobile-center-cli` supposed to be `appcenter-cli` ?

**Comment by Jamminroot** [comment]
Hey @kartikk and @mike-odom, thanks for bringing this to our sight. Please let us some time to investigate this on our side. We'll keep you updated in this issue.

**Comment by DmitriyKirakosyan** [comment]
Hi guys ,

This seems to be an issue with accessing to keychain.
Could you please try to apply this https://github.com/microsoft/appcenter/issues/1210#issuecomment-737446743 and let me know if it helps.

@mike-odom thank you for the investigation! The command you mentioned uses the old service name "mobile-center-cli" indeed, however it is not related to the problem. It is just a fallback to an old service name if the first command with "-s appcenter-cli" fails. See [profile.ts#L59](https://github.com/microsoft/appcenter-cli/blob/master/src/util/profile/profile.ts#L59) and [osx-token-store.ts#L65](https://github.com/microsoft/appcenter-cli/blob/master/src/util/token-store/osx/osx-token-store.ts#L65).
So basically it is a bug that the actual error gets swallowed by the fallback and you can't see what actually happened. I will add it to our board.

**Comment by mike-odom** [comment]
It's working well now! Though, I'm not sure if that fixed the issue. I had to unlock the keychain to get `appcenter login` to work prior to running `appcenter distribute`. And I think this was done when trying this command before. In between now and then the machine was rebooted to fix another issue with `codesign`, and that might have helped this as well.

**Comment by DmitriyKirakosyan** [comment]
Thank you @mike-odom for confirming that this problem is resolved for you.

Please feel free to reopen the issue if any additional help is required.

**Comment by akhil-x1** [comment]
Facing the same issue.

`Command 'codepush release-react -a profile-name/appname -d Staging' failed with exception "Command failed: /usr/bin/security find-generic-password -a profile-name -s mobile-center-cli -g
security: SecKeychainSearchCopyNext: The specified item could not be found in the keychain.`

**Comment by MikhailSuendukov** [comment]
Hello and thanks for reaching out to us! Have you tried the solution described in the thread?

**Comment by rajaa1272** [comment]
> Facing the same issue.
> 
> `appcenter login` does not create keychain entry for mobile-center-cli.
> 
> `Command 'codepush release-react -a profile-name/appname -d Staging' failed with exception "Command failed: /usr/bin/security find-generic-password -a profile-name -s mobile-center-cli -g security: SecKeychainSearchCopyNext: The specified item could not be found in the keychain.`

Hi @akhil-x1 have you fixed this?

**Comment by ngocht143** [comment]
still have this issue in v3.0.1

**Comment by akhilreddyt** [comment]
If above steps dont work, try removing appcenter profile `rm -rf ~/.appcenter-cli/profile.json`
https://stackoverflow.com/a/78946124/14372332
